test/image/mocks/bar_text_relative.json

```diff
@@ -0,0 +1,40 @@
+{
+    "data":[
+        {
+            "width":[1,0.8,0.6,0.4],
+            "text":[1,2,3,4],
+	    "textposition":"outside",
+            "y":[1,2,3,4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":[0.4,0.6,0.8,1],
+            "text":["Three",2,"This bar is size one",0],
+	    "textposition":"auto",
+            "y":[3,2,1,0],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":1,
+            "text":[-1,-3,-2,-4],
+	    "textposition":"inside",
+            "y":[-1,-3,-2,-4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "text":[0,"very small size",-3,-2],
```

Comment:
Yes, that's the correct behavior for inside labels, and it's basically the same thing we do for pies.

But there were two additional issues in my comment:
- for *outside* labels, shall we do the same thing? ie rotate the bar 90 degrees when it would allow us to enlarge the scaling?
- for *inside* labels, once we've figured out the maximum scaling, can we push the label over near the end of the bar rather than the middle?